File: .dist/happy.py
n=int(input("Enter a number:\n"))
sum=0

while n>0:
           rem=int(n%10)
           sum+=int(rem*rem)
           n=int(n/10)
           if n==0 and  sum>9 :
               n=sum
               sum=0

if sum==1:
       print("Happy Number")
else:        
      print("Not an happy Number")
    

File: .dist/happy.c
#include <stdio.h>

int main()
{
    printf("Enter a number:");
    int rem,sum=0,n;
    scanf("%d",&n);
      
        while(n>0)
        {
           rem=n%10;
           sum+=rem*rem;
           n=n/10;
           if(n==0 &&  sum>9) 
           {
            n=sum;
            sum=0;
           }            
        }

        if(sum==1)
        {
         printf("Happy Number");
        }
        else{
           printf("Not an happy Number");
        }

}